src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -136,17 +134,43 @@ mode with existing flows and hold flows of different table-mode types"() {
 
         then: "Flow remains valid and pingable, switch validation passes"
         northbound.validateFlow(flow.flowId).each { assert it.asExpected }
+        "validate switches"(involvedSwitches)
+        assert "is flow pingable?"(flow), FLOW_ISNT_PINGABLE_ASSERTION_MESSAGE
+
+        and: "Flow allows traffic"
+        assert "does flow pass traffic?"(flow), FLOW_DOESNT_PASS_TRAFFIC_ASSERTION_MESSAGE
+
+        when: "Delete flow"
+        def flowCookies = getFlowCookies(swPair, flow)
+        flowHelperV2.deleteFlow(flow.flowId)
+        def flowIsDeleted = true
+
+        then: "Flow rules are deleted from switches"
         involvedSwitches.each { sw ->
-            with(northbound.validateSwitch(sw.dpId)) { validation ->
-                validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
-                validation.verifyHexRuleSectionsAreEmpty(["missingHex", "excessHex", "misconfiguredHex"])
-                validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            with(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
+                rules.findAll { it.cookie in flowCookies }.empty
             }
         }
-        verifyAll(northbound.pingFlow(flow.flowId, new PingInput())) {
-            it.forward.pingSuccess
-            it.reverse.pingSuccess
-        }
+
+        cleanup: "Revert system to original state"
+        !flowIsDeleted && flow && flowHelperV2.deleteFlow(flow.flowId)
+        initSwProps && revertSwitchesToInitState(involvedSwitches, initSwProps)
+        northbound.deleteLinkProps(northbound.getLinkProps(topology.isls))
+    }
+
+    @Tidy
+    @Tags([LOW_PRIORITY])
+    def "System can hold flows of different table-mode types"() {
+        given: "A potential flow on a path of 4 switches: multi -> single -> multi -> single"
+        def (involvedSwitches, swPair, desiredPath, initSwProps) =
+        "get a potential flow on a path of 4 switches: multi and single"()
+        when: "Create the prepared hybrid flow"
```

Comment:
I think it's a good practice to have empty line between `geven`, `when`, `then`, `and`, 'cleanup'

Reply:
In fact, I believe that empty lines inside methods are the marks of bad design. Developer feels, that the method parts do several different things and tries to split the ones by empty line. Instead, in this case, one should split the method to several submethods with clear names.
But as long as it is convention of our project, I'll preserve the empty lines between steps.

Reply:
Not only of out project. Spock framework itself divides blocks with empty lines 
https://spockframework.org/spock/docs/1.1/all_in_one.html 
section: Specifications as Documentation

Reply:
Yes, unneeded empty lines is a common bad practice, not only in groovy or spock.